nilearn/maskers/surface_labels_masker.py

```diff
@@ -234,10 +195,55 @@ def __init__(
         self.clean_args = clean_args
 
     @property
-    def _labels_data(self):
-        """Return data of label image concatenated over hemispheres."""
-        all_labels = [x.ravel() for x in self.labels_img.data.parts.values()]
-        return np.concatenate(all_labels)
+    def n_elements_(self) -> int:
+        """Return number of regions.
+
+        This is equal to the number of unique values
+        in the fitted label image,
+        minus the background value.
+        """
+        check_is_fitted(self)
+        lut = self.lut_
+        return len(lut[lut["index"] != self.background_label])
```

Comment:
This is slightly different then the previous definition. Should they be the same, if yes is it possible to define this in a base class just once?

Reply:
yeah I have had to change things to make sure that behavior matched that of nifti label masker

Reply:
extracting in a parent class: I wanted to but n_elements is always a bit different from masker to masker (number of ROIs, number of voxels, number of maps...)